File: string_problems/FindWordsContainingCharacter.java

// Problem link : https://leetcode.com/problems/find-words-containing-character/description/?envType=problem-list-v2&envId=string&status=TO_DO&difficulty=EASY

package string_problems;

import java.util.ArrayList;
import java.util.List;

public class FindWordsContainingCharacter {
    public static List<Integer> findWordsContaining(String[] words, char x) {
        List<Integer> res = new ArrayList<>();

        // iterate through each word in the array
        for(int i=0; i<words.length; i++){
            String currWord = words[i];

            // check if the current word contains the character x
            if(currWord.indexOf(x) != -1){
                res.add(i);
            }
        }

        // return the list of indices where words contain the character x
        return res;
    }

    public static void main(String[] args) {
        // String[] words = {"leet","code"};
        // char x = 'e';

        // String[] words = {"abc","bcd","aaaa","cbc"};
        // char x = 'a';

        String[] words = {"abc","bcd","aaaa","cbc"};
        char x = 'z';
        System.out.println(findWordsContaining(words, x));
    }
}

File: string_problems/FindWordsContainingCharacter.js

// Problem link : https://leetcode.com/problems/find-words-containing-character/description/?envType=problem-list-v2&envId=string&status=TO_DO&difficulty=EASY

let findWordsContaining = function(words, x) {
    // initialize an empty array to store indices of matching words
    let res = [];

    // iterate through each word in the words array
    for(let i=0; i<words.length; i++){
        // get the current word at index i
        let currWord = words[i];

        // check if the current word contains the character x
        if(currWord.includes(x)){
            // if it does, add the index to the result array
            res.push(i);
        }
    }

    // return the array of indices where words contain the character x
    return res;
};

// let words = ["leet","code"], x = "e";

// let words = ["abc","bcd","aaaa","cbc"], x = "a";

let words = ["abc","bcd","aaaa","cbc"], x = "z"
console.log(findWordsContaining(words, x));
